### ISSUE TITLE
Unable to resolve `google-github-actions/setup-gcloud@master`

### ISSUE THREAD

**Issue by humbertowoody** [google-github-actions/setup-gcloud#451]
### TL;DR

Hi!

Our pipelines broke because of this change, we assume is just switching from `master` to `main`. However, we wanted to ask if this change is final? And why release this change 2 days before Christmas? :(

This broke all our pipelines and really didn't wanted to update everything given the holidays. Perhaps there's an easy fix we can apply instead of revisiting all of them? Thanks in advance!

### Expected behavior

To setup the action as normally.

### Observed behavior

The branch name has changed from `master` to `main` so it can't be loaded.

### Action YAML

```yaml
- name: Set GCP service account
        uses: google-github-actions/setup-gcloud@master
        with:
          service_account_key: ${{ secrets.GCP_SA_KEY }}
          export_default_credentials: true
```
```


### Additional information

This change impacts all our workflows (90+), if this change is a must, we can do it, but just to corroborate with you guys that it is indeed the expected approach.

**Comment by sethvargo** [comment]
According to https://github.blog/changelog/2020-07-22-github-actions-better-support-for-alternative-default-branch-names/, the rename should have been a non-breaking change. We'll revert this.

**Comment by sethvargo** [comment]
However, we do recommend pinning to `@v0` or a specific version.

**Comment by fatih** [comment]
This change also just broke our pipeline with the error:

```
Error: Unable to resolve action `google-github-actions/setup-gcloud@master`, unable to find version `master`
```

We fixed it by changing the branch to `main`.

**Comment by sethvargo** [comment]
@fatih we just reverted the change back to `master` to prevent existing workflows from breaking. We recommend pinning to `@v0` or a specific tag version, but we also didn't intend to introduce a breaking change with this. GitHub claimed renaming the branch would [automatically handle redirects](https://github.blog/changelog/2020-07-22-github-actions-better-support-for-alternative-default-branch-names/), but that appears to be inaccurate 😦

**Comment by humbertowoody** [comment]
Thank you very much @sethvargo! We'll wait for the revert; we were also aware of redirects and sorry for the inaccurate title.

**Comment by sethvargo** [comment]
The revert is done

**Comment by sethvargo** [comment]
@humbertowoody the revert is done. I would recommend scheduling time in the new year to either:

1. Pin to a specific version (`@v0.3.0`) or a floating pointer (`@v0`).

2. Switch to `google-github-actions/auth` if you're just performing authentication and not actually using the `gcloud` CLI.

**Comment by basit511** [comment]
I have tried everything from the above mentioned comments but still getting this error :
![Screenshot 2023-12-28 151441](https://github.com/google-github-actions/setup-gcloud/assets/101107213/25be8858-7c72-471b-afcf-77875f6d8e29)

**Comment by sethvargo** [comment]
Hi @basit511 - please open a new issue and complete the bug template that appears.
